src/crab/ebpf_domain.cpp

```diff
@@ -1485,9 +1565,19 @@ void ebpf_domain_t::operator()(const Bin& bin) {
                     havoc_offsets(bin.dst);
                 } else {
                     sub_overflow(dst.value, src.value);
-                    sub(get_type_offset_variable(bin.dst).value(), src.value);
-                    if (auto dst_offset = get_type_offset_variable(bin.dst))
+                    if (auto dst_offset = get_type_offset_variable(bin.dst)) {
                         sub(dst_offset.value(), src.value);
+                        if (type_inv.has_type(m_inv, dst.type, T_STACK)) {
```

Comment:
For code coverage, I added a test for this case, but the test just reproduces the issue in https://github.com/vbpf/ebpf-verifier/issues/321